src/toil/fileStore.py

```diff
@@ -851,7 +851,8 @@ def cleanCache(self, newJobReqs):
                 #                  (self.decodedFileID(cachedFile), cachedFileSize))
                 logger.debug('CACHE: Evicted  file with ID \'%s\' (%s bytes)' %
                              (self.decodedFileID(cachedFile), cachedFileSize))
-            assert cacheInfo.isBalanced(), 'Unable to free up enough space for caching.'
```

Comment:
This assertion is rather important. It usually happens because the job requirements are not accurate. We might need to add some explanation to the assertion message.

Pinging @arkal. We might need to change the assertion message to include a few hints as to possible reasons. If user error can lead to the assertion failure, it shouldn't be an assertion. Use require() from BPL.